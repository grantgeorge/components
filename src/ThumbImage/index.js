// @flow

/**
 *
 * ThumbImage
 *
 */

import React from 'react'
import styled from 'styled-components'

const ThumbImage = styled.div`
  background-color: #ccc;
  background-image: url(${(props) => props.imageUrl});
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  width: ${(props) => (props.width ? props.width : '100%')};
  height: ${(props) => (props.height ? props.height : '100%')};
  border-radius: 4px;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.1), 0 2px 3px rgba(0, 0, 0, 0.2);

  min-width: ${(props) => (props.minWidth ? props.minWidth : 'auto')};
  min-height: ${(props) => (props.minHeight ? props.minHeight : 'auto')};
`

type Props = {
  className: {},
  style: {},
  props: {},
}

const Component = ({ className, style, ...props }: Props) => {
  return <ThumbImage className={className} style={style} {...props} />
}

export default Component
