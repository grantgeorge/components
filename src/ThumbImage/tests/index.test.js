import React from 'react'
import { shallow } from 'enzyme'

import ThumbImage from '../index'

describe('<ThumbImage />', () => {
  let renderedComponent

  beforeEach(() => {
    renderedComponent = shallow(<ThumbImage />)
  })

  it('should render properly', () => {
    expect(renderedComponent).toMatchSnapshot()
  })
})
