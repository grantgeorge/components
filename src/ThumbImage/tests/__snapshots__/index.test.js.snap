// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThumbImage /> should render properly 1`] = `<styled.div />`;
